[UnifiedPDF] Going from Single Page to Two Pages doesn't reset the scroll position

smfr opened the pull request:
#### 30a915b5c386b5ecfc81b82be6d52072e8e6d108
<pre>
[UnifiedPDF] Going from Single Page to Two Pages doesn&apos;t reset the scroll position
<a href="https://bugs.webkit.org/show_bug.cgi?id=266610">https://bugs.webkit.org/show_bug.cgi?id=266610</a>
<a href="https://rdar.apple.com/119520473">rdar://119520473</a>

Reviewed by Tim Horton.

When changing from the single page to the 2-page view in the UnifiedPDF viewer, the document can get shorter,
so we may have to adjust the scroll position.

Do this after PDF layout. In order to set the scroll position, we need to implement `requestScrollToPosition()`
to talk to the relevant ScrollingCoordinator.

Not testable because you can&apos;t change the view mode in a test.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::updateScrollingExtents):
(WebKit::UnifiedPDFPlugin::requestScrollToPosition):
(WebKit::toWebCoreCursorType):

Canonical link: <a href="https://commits.webkit.org/272279@main">https://commits.webkit.org/272279@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bd4febec502d2e4c0e524dce85b67ab4d9d1e9da

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31122 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9794 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32810 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33628 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/28090 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12138 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7053 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27923 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31454 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/8253 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27817 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7088 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7257 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34967 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28321 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28169 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/33394 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7298 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5352 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/31232 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8986 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7333 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8001 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7833 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
```diff
@@ -455,6 +466,22 @@
     m_scrollContainerLayer->setEventRegion(WTFMove(eventRegion));
 }
 
+bool UnifiedPDFPlugin::requestScrollToPosition(const ScrollPosition& position, const ScrollPositionChangeOptions& options)
```
Should the other code (zooming) that requests a scroll make use of this instead of going straight to scrollingcoordinator?

smfr replied:
Probably. Will go find it (we have a bug about zooming).

hortont424 replied:
setPageScaleFactor just above

smfr committed:
[UnifiedPDF] Going from Single Page to Two Pages doesn't reset the scroll position
https://bugs.webkit.org/show_bug.cgi?id=266610
rdar://119520473

Reviewed by Tim Horton.

When changing from the single page to the 2-page view in the UnifiedPDF viewer, the document can get shorter,
so we may have to adjust the scroll position.

Do this after PDF layout. In order to set the scroll position, we need to implement `requestScrollToPosition()`
to talk to the relevant ScrollingCoordinator.

Not testable because you can't change the view mode in a test.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::updateScrollingExtents):
(WebKit::UnifiedPDFPlugin::requestScrollToPosition):
(WebKit::toWebCoreCursorType):

Canonical link: https://commits.webkit.org/272279@main
